hi-ml-histopathology/src/histopathology/configs/classification/BaseMIL.py

```diff
@@ -0,0 +1,111 @@
+It is responsible for instantiating the encoder and full DeepMIL model. Subclasses should define
+their datamodules and configure experiment-specific parameters.
+"""
+from pathlib import Path
+from typing import Type  # noqa
+
+import param
+from torch import nn
+from torchvision.models import resnet18
+
+from health_ml.lightning_container import LightningContainer
+from health_ml.networks.layers.attention_layers import AttentionLayer, GatedAttentionLayer
+
+from histopathology.datasets.base_dataset import SlidesDataset
+from histopathology.datamodules.base_module import CacheMode, CacheLocation, TilesDataModule
+from histopathology.models.deepmil import DeepMILModule
+from histopathology.models.encoders import (HistoSSLEncoder, IdentityEncoder,
+                                            ImageNetEncoder, ImageNetSimCLREncoder,
+                                            SSLEncoder, TileEncoder)
+
+
+class BaseMIL(LightningContainer):
+    # Model parameters:
+    pooling_type: str = param.String(doc="Name of the pooling layer class to use.")
+    # l_rate, weight_decay, adam_betas are already declared in OptimizerParams superclass
+
+    # Encoder parameters:
+    encoder_type: str = param.String(doc="Name of the encoder class to use.")
+    tile_size: int = param.Integer(224, bounds=(1, None), doc="Tile width/height, in pixels.")
+    n_channels: int = param.Integer(3, bounds=(1, None), doc="Number of channels in the tile.")
+
+    # Data module parameters:
+    batch_size: int = param.Integer(16, bounds=(1, None), doc="Number of slides to load per batch.")
+    max_bag_size: int = param.Integer(1000, bounds=(0, None),
+                                      doc="Upper bound on number of tiles in each loaded bag. "
+                                          "If 0 (default), will return all samples in each bag. "
+                                          "If > 0, bags larger than `max_bag_size` will yield "
+                                          "random subsets of instances.")
+    cache_mode: CacheMode = param.ClassSelector(default=CacheMode.MEMORY, class_=CacheMode,
+                                                doc="The type of caching to perform: "
+                                                    "'memory' (default), 'disk', or 'none'.")
+    precache_location: str = param.ClassSelector(default=CacheLocation.NONE, class_=CacheLocation,
+                                                 doc="Whether to pre-cache the entire transformed dataset upfront "
+                                                 "and save it to disk and if re-load in cpu or gpu. Options:"
+                                                 "`none` (default),`cpu`, `gpu`")
+    encoding_chunk_size: int = param.Integer(0, doc="If > 0 performs encoding in chunks, by loading"
+                                             "enconding_chunk_size tiles per chunk")
+    # local_dataset (used as data module root_path) is declared in DatasetParams superclass
```

Comment:
It would help to add here a standard environment path (or maybe it makes sense to add it in the hi-ml runner?). It's not something that is model specific